### DISCUSSION TITLE
Ability to drag and drop onto Mapsui?

### DISCUSSION THREAD

**Discussion by calicoder77** [Mapsui/Mapsui#2319] [Q&A] [answered]
Good evening, I have a WPF C# app that has 2 columns: left one display the Mapsui world map and the right-side is a table with rows I can drag and drop onto the left side map. Thing is nothing is display on the map itself, it's like the row item gets dropped back and map stays on top? Just wondering if it is possible to use the map in a WPF class to allow dragging and dropping items on it?

Thank you in advance for any tips or tricks. All the best!

**Comment by charlenni** [comment]
There is nothing special about this view. Mapsui uses SkiaSharp to draw the maps. So you should have all events, that SkiaSharp for WPF has.

**Reply by calicoder77** [reply]
Thank you for the quick reply. I see that the MouseOver function works well but not the others when I start to click and hold the mouse outside the Mapsui frame. For example: if I click the right side table and start to drag over to the left-side where Mapsui is located, only MouseOver works. No other mouse event gets triggered. I'd like to trigger the MouseLeftButtonReleased one preferably.

**Reply by pauldendulk** [reply, marked as answer]
I just tested this by adding a handler in Windows1.xaml.cs of the wpf sample and LeftMouseButtonUp is called.

![image](https://github.com/Mapsui/Mapsui/assets/963462/6b9bd051-98e6-4fdd-b9aa-c2f0ac9bfa81)

**Reply by calicoder77** [reply]
Thank you again.
